source/fundamentals/auth.txt

```diff
@@ -189,59 +189,104 @@ username and password to authenticate your user.
-   if err != nil {
-      panic(err)
-   }
-   _ = awsIAMClient
+.. important::
+   
+   The driver only reads the credentials from the first method that it detects
+   in the order as given by the preceding list. For example, if you specify 
+   your AWS credentials in the connection string, the driver ignores any 
+   credentials that you specified in environment variables.
 
-If you need to specify an AWS session token, use the temporary
-credentials returned from an assume role request.
+.. tabs::
 
-To use temporary credentials, assign the ``AuthMechanismProperties``
-option the value of your ``sessionToken``:
+   .. tab:: Connection String
+      :tabid: connection string
 
-.. code-block:: go
-   :emphasize-lines: 7-9
+      .. tip::
 
-   var sessionToken string
-   assumeRoleCredential := options.Credential{
-      AuthMechanism: "MONGODB-AWS",
-      AuthSource: "<authenticationDb>",
-      Username: "<accessKeyID>",
-      Password: "<secretAccessKey>",
-      AuthMechanismProperties: map[string]string{
-         "AWS_SESSION_TOKEN": "<sessionToken>",
-      },
-   }
-   assumeRoleClient, err := mongo.Connect(context.TODO(),
-      options.Client().SetAuth(assumeRoleCredential))
+         The following examples set the appropriate credentials using the ``SetAuth()`` 
+         method. You can also specify these credentials using the ``ApplyURI()`` 
+         method. If you use the ``ApplyURI()`` method you must URL encode the username 
+         and password to ensure they are correctly parsed.
+   
+      To connect to your MongoDB instance using your AWS IAM credentials, perform the
+      following steps:
+
+      - Assign the ``AuthMechanism`` option the value ``MONGODB-AWS``
+      - Assign the ``Username`` option the value of your ``accessKeyID``
+      - Assign the ``Password`` option the value of your ``secretAccessKey``
+
+      .. literalinclude:: /includes/fundamentals/code-snippets/authentication/aws-connection-string.go
+         :language: go
+
+      If you need to specify an AWS session token, use the temporary
+      credentials returned from an assume role request.
+      
+      To use temporary credentials, assign the ``AuthMechanismProperties``
+      option the value of your ``sessionToken``:
+
+      .. literalinclude:: /includes/fundamentals/code-snippets/authentication/aws-connection-string-session-token.go
+         :language: go
+
+   .. tab:: Environment Variables
+      :tabid: environment variables
+
+      To authenticate to your MongoDB instance using AWS credentials stored in
+      environment variables, use a shell to set the variables as follows:
+
+      .. code-block:: bash
+
+         export AWS_ACCESS_KEY_ID=<awsKeyId>
+         export AWS_SECRET_ACCESS_KEY=<awsSecretKey>
+         export AWS_SESSION_TOKEN=<awsSessionToken>
+
+      .. note::
+
+         Omit the line containing ``AWS_SESSION_TOKEN`` if you don't need an AWS
+         session token for that role.
```

Comment:
```suggestion
         If you don't need an AWS
         session token for that role, omit the line containing ``AWS_SESSION_TOKEN`` .
```

Also, what does "that" refer to in "that role"?

Reply:
Clarified "that" - let me know if it makes more sense now